### PR TITLE
Rent a house Complete Project (Final Capstone Project Group)

### PR DESCRIPTION
**Pull request by kessio**
Hello 👋🏾 
## Our group is repeating the Final Capstone Project so this is the design for our project:
https://github.com/microverseinc/curriculum-final-capstone/blob/main/projects/buisness_requirements_2.md

We are a team of 3 members @samuelmoiwa @ecekpo @kessio 

Our Front-end PR: (https://github.com/samuelmoiwa/rent-a-house-frontend/pull/33) 

## Project Implementations: 
- The user logs in to the website, only by typing the username (a proper authenticated login is a requirement if your group is made of 5 people).
- In the navigation panel, the user can see links to:
Houses
"My favorites".
"Add house
"Delete house
- On the main page, the user can see a list of houses
- When the user selects a specific item, they can see the details page with its full description (skip the "Rotate image" button).
- In the details page, the user can click the "Add to favourite" button 
- When the user clicks the "My favorites" link in the navigation panel they can see a list of their favorite items.
- When the user clicks the "Add item" link in the navigation panel they can see a form for adding a new item
- When the user clicks the "Delete item" link in the navigation panel they can see a list of all items with a title and "Delete" button.
- When the user clicks the "Delete" button, the selected item is marked as removed and does not show on the main list anymore.
- Make the app responsive, creating both mobile and desktop versions. IMPORTANT NOTE: there is no desktop design for this project, so create it on your own.
- Add full documentation for your API.

-

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+         :recoverable, :rememberable, :validatable,
+         :jwt_authenticatable, jwt_revocation_strategy: self
+
+  has_many :houses
+  has_many :favorites
+
+  def image_url
+    Rails.application.routes.url_helpers.url_for(image) if image.attached?
+  end
+end
```
Comment (mcrd25) [inline app/models/user.rb:15]: [OPTIONAL]

is there any validations you can implement for fields in your user model?

**Review by mcrd25 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good documentation 👍 
- Your routes look good enough 👍 
- Your controllers looks good enough 👍 
- Your models look good 👍 

## Required Changes ♻️
_Check the comments under the review._
- You must add tests. You can add using rspec
- Your README needs some cleaning up to look more professional
- Reference your Front-End PR is PR description

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [OPTIONAL] When doing tests, consider both positive and negative scenarios. what do you expect to happen when inputs are all good and what do you expect to happen when inputs are bad?


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Meltrust — CHANGES_REQUESTED** [review]
_note: The team claims they are repeating the module in their PR description.  Therefore, reviewing normally as a first review, as requested._

Hi team,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  Nice code organization ✔️ 
-  API is working well✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Comment by mcrd25** [comment]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- implemented changes to bst of ability 👍 

## Optional suggestions
- Remove the empty line to prevent the linter failure before merging

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ecekpo** [comment]
Done. Thanks

**Pull request merged**